torch/lib/c10d/FileStore.cpp

```diff
@@ -234,12 +257,17 @@ int64_t FileStore::add(const std::string& key, int64_t i) {
     auto len = value.size();
     ti += std::stoll(std::string(buf, len));
   }
-
+  // Always seek to the end to write
+  file.seek(0, SEEK_END);
```

Comment:
The file position will already be at the end of the file at this point. No need for SEEK_END, see `refresh`. This is guaranteed because the exclusive lock on the file is held.

Reply:
That's where the bug is, there is no guarantee that refresh will seek to the end.  And set will always move seek to the end, why not add, they are both the only two write paths.